tests/eessi_test_mapping/software_to_tests.yml

```diff
@@ -0,0 +1,35 @@
+# This file creates a mapping between (regular expressions for) module names and test names from the EESSI test suite
+# If a module name matches one of the regular expressions, the listed set of tests will be run in the test step
+# For a given module name, the test list for the first matching regular expression is returned
+# E.g. for
+# mappings:
+#   foo-v1:
+#     - bar
+#   foo-*
+#     - bar2
+# only the bar test will be run for foo-v1 (even though it also matches the pattern (foo-*)
+# If a module name does not match anything, the default_tests will be run
+# Note that to list all available tests by name, one can do execute
+# reframe -R -c /path/to/eessi/test-suite/ --list | grep -Po "\bEESSI_\S+?(?=[\s'])" | uniq
+# Note that this regular expression is a bit sensitive to changes in the structure of ReFrame's output,
+# but is confirmed to work for ReFrame version 4.6.1
+mappings:
+  PyTorch-Bundle/*:
+    - EESSI_PyTorch_torchvision
+  QuantumESPRESSO/*:
+    - EESSI_QuantumESPRESSO
+  CP2K/*:
+    - EESSI_CP2K
+  ESPResSo/*:
+    - EESSI_ESPRESSO
+  LAMMPS/*:
+    - EESSI_LAMMPS
+  OSU-Micro-Benchmarks/*:
+    - EESSI_OSU_Micro_Benchmarks
+  GROMACS/*:
+    - EESSI_GROMACS
```

Comment:
@casparvl We'll need to maintain this as we add more tests to the test suite, right?

Wouldn't it make more sense to make sure that we consistently use `EESSI_<software_name>`, so we don't need this mapping file at all?